From 9a0e4e773570de3777d0980bc004ef9d0a05c76c Mon Sep 17 00:00:00 2001
From: Cui GaoSheng <cuigaosheng1@huawei.com>
Date: Sun, 28 Mar 2021 23:13:04 -0400
Subject: [PATCH] arm32: kaslr: Fix the bug of symbols relocation
Patch-mainline: Not yet, from openEuler
References: bsn#22
openEuler-commit: 9a0e4e773570de3777d0980bc004ef9d0a05c76c


hulk inclusion
category: bugfix
bugzilla: 47952
CVE: NA

------------------------------------------------------------------------

Use adr_l instead of adr macro for symbol relocation, because linux
symbol relocation has scope restrictions.

Signed-off-by: Cui GaoSheng <cuigaosheng1@huawei.com>
Reviewed-by: Xiu Jianfeng <xiujianfeng@huawei.com>
Signed-off-by: Zheng Zengkai <zhengzengkai@huawei.com>
Signed-off-by: Guoqing Jiang <guoqing.jiang@suse.com>
---
 arch/arm/boot/compressed/head.S | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/arch/arm/boot/compressed/head.S b/arch/arm/boot/compressed/head.S
index 4b95a9268aba..1ba21b868ea3 100644
--- a/arch/arm/boot/compressed/head.S
+++ b/arch/arm/boot/compressed/head.S
@@ -152,7 +152,7 @@
 		 * in little-endian form.
 		 */
 		.macro	get_inflated_image_size, res:req, tmp1:req, tmp2:req
-		adr	\res, .Linflated_image_size_offset
+		adr_l	\res, .Linflated_image_size_offset
 		ldr	\tmp1, [\res]
 		add	\tmp1, \tmp1, \res	@ address of inflated image size
 
@@ -308,7 +308,7 @@ not_angel:
 		orrcc	r4, r4, #1		@ remember we skipped cache_on
 		blcs	cache_on
 
-restart:	adr	r0, LC1
+restart:	adr_l	r0, LC1
 		ldr	sp, [r0]
 		ldr	r6, [r0, #4]
 		add	sp, sp, r0
-- 
2.26.2

